src/CommonLib/Ntlm/HttpNtlmAuthenticationService.cs

```diff
@@ -0,0 +1,205 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Security.Authentication;
+using System.Threading.Tasks;
+
+namespace SharpHoundCommonLib.Ntlm;
+
+public class HttpNtlmAuthenticationService
+{
+    private readonly ILogger _logger;
+    private readonly IHttpClientFactory _httpClientFactory;
+    public Uri Url { get; set; }
+
+    public HttpNtlmAuthenticationService(ILogger logger, IHttpClientFactory httpClientFactory)
+    {
+        _logger = logger;
+        _httpClientFactory = httpClientFactory;
+    }
+
+    public async Task EnsureRequiresAuth(bool enableSigning, bool? useBadChannelBindings)
+    {
+        if (Url == null)
```

Comment:
It looks like the `Url` property is only used on this method while being passed in to our other methods, might be worth moving this to a required argument rather than a property